File: apps/mobile/lib/utils/text.dart
/// Capitalises the first letter of every word: "upward facing dog (male)" ->
/// "Upward Facing Dog (Male)". Leaves the rest of each word as-is.
String titleCase(String s) => s.replaceAllMapped(
      RegExp(r'\w+'),
      (m) => '${m[0]![0].toUpperCase()}${m[0]!.substring(1)}',
    );

File: apps/native-mobile/src/utils/text.ts
// Ported from apps/mobile/lib/utils/text.dart.
/** Capitalises the first letter of every word, leaving the rest as-is. */
export function titleCase(s: string): string {
  return s.replace(/\w+/g, (w) => w[0].toUpperCase() + w.substring(1));
}
